Use external shutdown service

aitorcas23 opened the pull request:
This pull request is based off of dockerize-config-file-handler branch.

This pull request changes the functionality of iombina-config-file-handler so it uses [iombian-shutdown-service](https://github.com/Tknika/iombian-shutdown-service).
This way, when running the docker container generated from this container it won't need to be privileged.

aitorcas23 committed:
Generate docker files and use environment variables

aitorcas23 committed:
Dockerized iombian-config-file-handler working

aitorcas23 committed:
Make internal button events port and host uncustomizable

aitorcas23 committed:
Get button events host and port from environment variables

aitorcas23 committed:
Use localhost as default button events host

aitorcas23 committed:
Change parameters.yml location inside container

aitorcas23 committed:
Remove docker-compose.yaml

aitorcas23 committed:
Add docker section to README

aitorcas23 committed:
Working with external shutdown service

aitorcas23 committed:
Update readme for using shutdown service

aitorcas23 committed:
Bold docker section list

aitorcas23 committed:
Add missing shutdown host and port to README

bodiroga commented on README.md:
```diff
@@ -51,6 +51,53 @@ The communication must be done through a ZeroMQ socket (5555 port by default), f
+The passed volume is used to access the machine configuration from the container.
+
+- **-e** can be used to define the environment variables:
+    - CONFIG_PORT: the port where the services will connect to acces the configuration.
```
"access", right? 😉

bodiroga commented on README.md:
```diff
@@ -51,6 +51,53 @@ The communication must be done through a ZeroMQ socket (5555 port by default), f
+- **-e** can be used to define the environment variables:
+    - CONFIG_PORT: the port where the services will connect to acces the configuration.
+    Default value is 5555.
+    - RESET_EVENT: the button event to reset the raspberry.
```
```suggestion
    - RESET_EVENT: the button event that should trigger the reset of the IoMBian device
```

bodiroga commented on README.md:
```diff
@@ -51,6 +51,53 @@ The communication must be done through a ZeroMQ socket (5555 port by default), f
+    Default value is 5556.
+    - YAML_FILE_PATH: The path where the parameters.yml file is located inside the container.
+    Default path is /app/parameters.yml.
+    - SHUTDOWN_HOST: The host where the reboot event will be sent.
```
```suggestion
    - SHUTDOWN_HOST: The host where the reboot command will be sent.
```

Recuerda que el iombian-shutdown-service emplea el patrón request-response, por lo que lo que se le envía es un comando, no un evento 😉

aitorcas23 replied:
Por lo que he visto command se utiliza cuando le envias un mensaje pidiendo que haga algo, y event es más para cuando ocurre algo y lo quieres comunicar, como por ejemplo en button-handler. Tomo nota.

bodiroga commented on README.md:
```diff
@@ -51,6 +51,53 @@ The communication must be done through a ZeroMQ socket (5555 port by default), f
+    Default path is /app/parameters.yml.
+    - SHUTDOWN_HOST: The host where the reboot event will be sent.
+    Default value is "127.0.0.1"
+    - SHUTDOWN_PORT: The port where the reboot event will be sent.
```
```suggestion
    - SHUTDOWN_PORT: The port where the reboot command will be sent.
```

Recuerda que el iombian-shutdown-service emplea el patrón request-response, por lo que lo que se le envía es un comando, no un evento 😉

bodiroga commented on src/main.py:
```diff
@@ -32,8 +42,10 @@ def signal_handler(sig, frame):
+    comm_module.execute_command(RESET_COMMAND)
     stop()
-    os.system('reboot')
+    # os.system('systemctl reboot')
```
Remove the commented line 😉

bodiroga commented on src/main.py:
```diff
@@ -32,8 +42,10 @@ def signal_handler(sig, frame):
 def config_update_callback():
     logger.info("Rebooting the system")
+    logger.debug(f"Sending {RESET_COMMAND} command to {SHUTDOWN_HOST}:{SHUTDOWN_PORT}")
+    comm_module.execute_command(RESET_COMMAND)
```
Shouldn't this function be called after the ```stop()```command?

aitorcas23 replied:
The stop command stops all the ZeroMQ clients, so if you call stop() before comm_module.execute_command() it will give an error. I could separate the stop function into pieces so the first one stops all except the comm_module and then, after comm_module.execute_command(), stop the comm_module.

bodiroga replied:
You are totally right, @aitorcas23, leave it as it is 😉

aitorcas23 committed:
Fix suggested changes #6

bodiroga reviewed (approved):
LGTM! 👍🏻

bodiroga commented on README.md:
```diff
@@ -51,6 +51,53 @@ The communication must be done through a ZeroMQ socket (5555 port by default), f
+
+After building the image, execute it with docker run
+
+`docker run --name ${CONTAINER_NAME} --rm -d -p 5555:5555 -v /boot/config/parameters.yml:/app/parameters.yml -e RESET_EVENT=tripe_click`
```
@aitorcas23, me acabo de dar cuenta de que aquí falta el nombre de la imagen, ¿no?

```suggestion
`docker run --name ${CONTAINER_NAME} --rm -d -p 5555:5555 -v /boot/config/parameters.yml:/app/parameters.yml -e RESET_EVENT=tripe_click iombian-config-file-handler:latest`
```

bodiroga commented on src/main.py:
```diff
@@ -4,26 +4,36 @@
+SHUTDOWN_HOST = os.environ.get("SHUTDOWN_HOST", "127.0.0.1")
+SHUTDOWN_PORT = int(os.environ.get("SHUTDOWN_PORT", 5558))
+
+PUBLISHER_HOST = "0.0.0.0"
```
@aitorcas23, esto se me había escapado, pero: ¿publisher_host? Este servicio es el que se usa en los otros microservicios para obtener información del fichero parameters.yml, ¿no? Creo que la referencia a un 'publisher' no es correcta. ¿CONFIG_HOST? ¿CONFIG_LISTENER_HOST? Diría que cualquier me vale 👍🏻

aitorcas23 committed:
Apply requested changes #6

- Add image to example docker run command
- Rename publlisher host to config host

aitorcas23 committed:
Rename shutdown host and port

Rename shutdown host and port to shutdow service host and port for consistency with other services.

aitorcas23 committed:
Fix previous commit mistake

bodiroga reviewed (approved):
LGTM! 👍🏻
